engine/src/main/java/org/destinationsol/game/Console.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game;
+
+import com.badlogic.gdx.Input;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.math.Vector2;
+import org.destinationsol.SolApplication;
+import org.destinationsol.assets.Assets;
+import org.destinationsol.common.SolColor;
+import org.destinationsol.common.SolException;
+import org.destinationsol.game.console.ConsoleInputHandler;
+import org.destinationsol.game.console.ShellInputHandler;
+import org.destinationsol.ui.SolInputManager;
+import org.destinationsol.ui.SolUiControl;
+import org.destinationsol.ui.SolUiScreen;
+import org.destinationsol.ui.UiDrawer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Console implements SolUiScreen {
+    private static final Vector2 TOP_LEFT = new Vector2(0.03f, 0.03f);
+    private static final Vector2 BOTTOM_RIGHT = new Vector2(0.8f, 0.5f);
+    private static final float FRAME_WIDTH = 0.02f;
+    private static final int MAX_WIDTH_OF_LINE = 1040;
+    private static Console instance;
+    private final BitmapFont font;
+    private final SolUiControl exitControl;
+    private final List<String> linesOfOutput;
+    private final List<SolUiControl> controls;
+
+    private StringBuilder inputLine;
+    private ConsoleInputHandler inputHandler;
+
+    public void setInputHandler(ConsoleInputHandler inputHandler) {
+        this.inputHandler = inputHandler;
+    }
+
+    private boolean isActive;
+
+    private Console() {
+        font = Assets.getFont("engine:main").getBitmapFont();
+        exitControl = new SolUiControl(null, true, Input.Keys.ESCAPE);
+        controls = Collections.singletonList(exitControl);
+        linesOfOutput = new ArrayList<>();
+        inputLine = new StringBuilder();
+        println("Welcome to the world of Destination Sol! Your journey begins!");
+        setInputHandler(new ShellInputHandler());
+    }
+
+    public void println(String s) {
```

Comment:
Perhaps also javadoc here. Curiously have you checked out the Terasology console classes for some of the potential there, like auto-completion, documented parameters, etc?